drivers/rtc/rtc_ll_stm32.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2023 Prevas A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#define DT_DRV_COMPAT st_stm32_rtc
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/clock_control.h>
+#include <soc.h>
+#include <stm32_ll_pwr.h>
+#include <stm32_ll_rcc.h>
+#include <stm32_ll_rtc.h>
+#include <stm32_hsem.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(rtc_stm32, CONFIG_RTC_LOG_LEVEL);
+
+/* Convert calendar start time */
+/* RTC start time: 1st, Jan, 2000 */
+/* struct tm start:   1st, Jan, 1900 */
+#define TM_TO_RTC_OFFSET 100
+
+/* clang-format off */
+
+#if defined(CONFIG_SOC_SERIES_STM32L4X)
+#define RTC_EXTI_LINE	LL_EXTI_LINE_18
+#elif defined(CONFIG_SOC_SERIES_STM32C0X) \
+	|| defined(CONFIG_SOC_SERIES_STM32G0X)
+#define RTC_EXTI_LINE	LL_EXTI_LINE_19
+#elif defined(CONFIG_SOC_SERIES_STM32F4X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F0X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F2X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F3X) \
+	|| defined(CONFIG_SOC_SERIES_STM32F7X) \
+	|| defined(CONFIG_SOC_SERIES_STM32WBX) \
+	|| defined(CONFIG_SOC_SERIES_STM32G4X) \
+	|| defined(CONFIG_SOC_SERIES_STM32L0X) \
+	|| defined(CONFIG_SOC_SERIES_STM32L1X) \
+	|| defined(CONFIG_SOC_SERIES_STM32L5X) \
+	|| defined(CONFIG_SOC_SERIES_STM32H7X) \
+	|| defined(CONFIG_SOC_SERIES_STM32H5X) \
+	|| defined(CONFIG_SOC_SERIES_STM32WLX)
+#define RTC_EXTI_LINE	LL_EXTI_LINE_17
+#endif
+
+/* clang-format on */
+
+#if defined(CONFIG_SOC_SERIES_STM32F1X)
+#error RTC not supported for STM32F1X
```

Comment:
From what I can see the F1 series uses different functions as all other series, but still has an RTC. I can understand that this first version of the driver has no support for F1, but I wonder what is the impact of this. If a user is trying to use the RTC the old way (as a counter) does this error occur? If so, that is a problem.
(Furthermore, there would be plenty of other compilation errors beside this one, since many LL functions used here don't exist on F1.)

Reply:
I do understand your confusion. I see I'm missing a commit, where COUNTER and RTC are made mutual exclusive